src/v1/sources/shopify/webhookTransformations/serverSideTransform.js

```diff
@@ -0,0 +1,193 @@
+  message.setEventName(SHOPIFY_TRACK_MAP[shopifyTopic]);
+  // eslint-disable-next-line camelcase
+  const { line_items: lineItems } = event;
+  const productsList = getProductsFromLineItems(lineItems, lineItemsMappingJSON);
+  message.setProperty('properties.products', productsList);
+  return message;
+};
+
+const processEvent = async (inputEvent, metricMetadata) => {
+  let message;
+  const event = lodash.cloneDeep(inputEvent);
+  let redisData;
+  const shopifyTopic = getShopifyTopic(event);
+  delete event.query_parameters;
+  switch (shopifyTopic) {
+    case IDENTIFY_TOPICS.CUSTOMERS_CREATE:
+    case IDENTIFY_TOPICS.CUSTOMERS_UPDATE:
+      message = identifyPayloadBuilder(event);
+      break;
+    case ECOM_TOPICS.ORDERS_CREATE:
+    case ECOM_TOPICS.ORDERS_UPDATE:
+    case ECOM_TOPICS.CHECKOUTS_CREATE:
+    case ECOM_TOPICS.CHECKOUTS_UPDATE:
+      message = ecomPayloadBuilder(event, shopifyTopic);
+      break;
+    case 'carts_update':
+      if (useRedisDatabase) {
+        redisData = await getDataFromRedis(event.id || event.token, metricMetadata, 'Cart Update');
+        const isValidEvent = await checkAndUpdateCartItems(
+          inputEvent,
+          redisData,
+          metricMetadata,
+          shopifyTopic,
+        );
+        if (!isValidEvent) {
+          return NO_OPERATION_SUCCESS;
+        }
+      }
+      message = trackPayloadBuilder(event, shopifyTopic);
+      break;
+    default:
+      if (!SUPPORTED_TRACK_EVENTS.includes(shopifyTopic)) {
+        stats.increment('invalid_shopify_event', {
+          writeKey: metricMetadata.writeKey,
+          source: metricMetadata.source,
+          shopifyTopic: metricMetadata.shopifyTopic,
+        });
+        return NO_OPERATION_SUCCESS;
+      }
+      message = trackPayloadBuilder(event, shopifyTopic);
+      break;
+  }
+
+  if (message.userId) {
+    message.userId = String(message.userId);
+  }
+  if (!get(message, 'traits.email')) {
+    const email = extractEmailFromPayload(event);
+    if (email) {
+      message.setProperty('traits.email', email);
+    }
+  }
+  if (message.type !== EventType.IDENTIFY) {
+    const { anonymousId } = await getAnonymousIdAndSessionId(
+      message,
+      { shopifyTopic, ...metricMetadata },
+      redisData,
+    );
+    if (isDefinedAndNotNull(anonymousId)) {
+      message.setProperty('anonymousId', anonymousId);
+    }
+  }
+  message.setProperty(`integrations.${INTEGERATION}`, true);
+  message.setProperty('context.library', {
+    eventOrigin: 'server',
+    name: 'RudderStack Shopify Cloud',
+    version: '2.0.0',
+  });
+  message.setProperty('context.topic', shopifyTopic);
+  // attaching cart, checkout and order tokens in context object
+  message.setProperty(`context.cart_token`, event.cart_token);
+  message.setProperty(`context.checkout_token`, event.checkout_token);
+  // raw shopify payload passed inside context object under shopifyDetails
+  message.setProperty('context.shopifyDetails', event);
```

Comment:
raw event name from shopify should also be maintained in 'context.shopifyDetails'